controllers/UserController.ts

```diff
@@ -107,9 +109,35 @@ export default class UserController implements UserControllerI {
      * @param {Response} res Represents response to client, including status
      * on whether updating a user was successful or not
      */
-    updateUser = (req: Request, res: Response) =>
-        UserController.userDao.updateUser(req.params.uid, req.body)
-            .then(status => res.json(status))
+    updateUser = async (req: Request, res: Response) => {
+        const userInfo = req.body
+        let existingUser = false
+
+        // checks if username is taken
+        if (userInfo.username) {
```

Comment:
If a user doesn't update their username, this checking could always fail.
Perhaps we should compare the username from current session.

Reply:
Since I am send all profile to the backend. Thanks